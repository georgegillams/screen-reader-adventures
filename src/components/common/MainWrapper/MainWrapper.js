import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';

import STYLES from './main-wrapper.scss';
import { DEBUG_SHOW_PAGE_CONTAINER_KEY } from 'helpers/storageConstants';

const getClassName = cssModules(STYLES);

const MainWrapper = props => {
  const { className, ...rest } = props;
  const [showDebug, setShowDebug] = useState(false);

  useEffect(() => {
    setShowDebug(window.localStorage.getItem(DEBUG_SHOW_PAGE_CONTAINER_KEY) === 'true');
  }, []);

  const outerClassNames = [getClassName('page-container__container')];
  if (showDebug) {
    outerClassNames.push([getClassName('page-container__container--debug')]);
  }

  if (className) {
    outerClassNames.push(className);
  }

  return <main id="main" className={outerClassNames.join(' ')} {...rest} />;
};

MainWrapper.propTypes = {
  className: PropTypes.string,
};

MainWrapper.defaultProps = {
  className: null,
};

export default MainWrapper;
