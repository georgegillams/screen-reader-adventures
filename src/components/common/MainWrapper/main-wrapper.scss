@import '~utils/tokens';

.main-wrapper {
  &__container {
    width: 100%;

    &--debug {
      background-color: orange;

      @media (prefers-color-scheme: dark) {
        background-color: darkorange;
      }
    }
  }
}
