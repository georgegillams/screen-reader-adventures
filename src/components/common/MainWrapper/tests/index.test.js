import React from 'react';

import { render } from '@testing-library/react';

import MainWrapper from '..';

describe('<MainWrapper />', () => {
  const children = <div>some text</div>;

  it('Should render correctly', () => {
    const { container } = render(<MainWrapper>{children}</MainWrapper>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(<MainWrapper className={'test-custom-classname'}>{children}</MainWrapper>);

    expect(container).toMatchSnapshot();
  });
});
