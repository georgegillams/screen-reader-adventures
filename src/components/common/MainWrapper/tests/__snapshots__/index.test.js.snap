// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainWrapper /> Should render correctly 1`] = `
<div>
  <main
    class="page-container__container"
    id="main"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;

exports[`<MainWrapper /> Should render correctly with custom className 1`] = `
<div>
  <main
    class="page-container__container test-custom-classname"
    id="main"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;
